doc/barclamp.md

```diff
@@ -1356,3 +1356,147 @@ place) you will have to add additional setup code to
 and/or
 [qa_crowbarsetup.sh](https://github.com/SUSE-Cloud/automation/blob/master/scripts/qa_crobarsetup.sh)
 to take care of your barclamp's needs.
+
+## High Availability (HA) Integration
+
+Using the [barbican barclamp](https://github.com/crowbar/crowbar-openstack/tree/9805c7ddb81ffbf6f2b79061c6ceb332659cd614/chef/cookbooks/barbican)
+as an example. We will try to investigate what changes are required to make a barclamp
+HA compatible
+
+[This commit](https://github.com/crowbar/crowbar-openstack/pull/512/files) is
+the first place we start looking. It was not entirely correct and needed
+[more commits](https://github.com/crowbar/crowbar-openstack/commits/master/chef/cookbooks/barbican)
+to correct.
+
+-  [default.rb](https://github.com/crowbar/crowbar-openstack/commits/master/chef/cookbooks/barbican/attributes/default.rb)
+
+   This file defines the default for all the values of the vars required by barclamp.
+It allows you to set defaults (or overrides) for any attributes in the node object.
+In this case we will use that capability to set a bunch of HA related attributes that
+are rarely changed an, hence need not be put int the data bag.
+
+```ruby
+default[:barbican][:ha]
+
+# HA attributes
+   [:enabled] = false
+
+# Ports to bind to when haproxy is used for the real ports
+   [:ports][:api] = 5621
+
+# pacemaker definitions
+   [:api][:op][:monitor][:interval] = "10s"
+   [:worker][:op][:monitor][:interval] = "10s"
+   [:keystone_listener][:op][:monitor][:interval] = "10s"
+
+   [:worker][:agent] = "systemd:openstack-barbican-worker"
+   [:keystone_listener][:agent] = "systemd:openstack-barbican-keystone-listener"
+
+```
+> - Resource Agent: In 'LayMan' terms, resource agent is a script that the
+    cluster manager uses to manage the resource.
+> - LSB: is a specification of writing resource agent. 
+> - OCF: Pacemaker specific extension to LSB.
+> - systemd: Use systemd unit files for managing the resource.
+
+- [barbican_service.rb(cookbook)](https://github.com/crowbar/crowbar-openstack/commits/master/chef/cookbooks/barbican/definitions)
+
+   Makes the installation of the service HA aware, by using [```provider  Chef::Provider::CrowbarPacemakerService```](https://github.com/crowbar/crowbar-ha/blob/master/chef/cookbooks/crowbar-pacemaker/providers/service.rb)
```

Comment:
This paragraph can probably be condensed a bit. `Chef::Provider::CrowbarPacemakerService` is essentially a dummy provider that does nothing and is used for the _regular_ service resource in the HA case: this ensures crowbar does enable/start the service because that is up to the Pacemaker resource agent in this case.

Also, the Barbican barclamp in its current shape does not use it, because it only runs services that are horizontally scalable anyway (such as barbican-worker). There are some vestiges in [definitions/barbican.rb](https://github.com/crowbar/crowbar-openstack/blob/f26dc78fe3038ebb5bdb426e70d6750f2351cf29/chef/cookbooks/barbican/definitions/barbican_service.rb), but nothing uses that anymore.

You'll find `CrowbarPacemakerService` in use in the Neutron barclamp, for example: https://github.com/crowbar/crowbar-openstack/blob/f26dc78fe3038ebb5bdb426e70d6750f2351cf29/chef/cookbooks/neutron/recipes/server.rb#L374